### PR TITLE
Performance: Add preallocation for some slices

### PR DESCRIPTION
**Pull request by peakle**
**What is this feature?**

Add preallocations for slices to decrease number of calls to the go allocator / OS api and thereby reduce the program execution time. It's little part of https://github.com/grafana/grafana/pull/59118 PR.

**Why do we need this feature?**

It's perfomance improvement.

**Who is this feature for?**

\-

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by peakle** [comment]
@papagian who can help me with review?

File: pkg/services/alerting/notifiers/googlechat.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func (gcn *GoogleChatNotifier) Notify(evalContext *alerting.EvalContext) error {
 	}
 
 	// add a text paragraph widget for the fields
+	//nolint:prealloc // break block
```
Comment (papagian) [inline pkg/services/alerting/notifiers/googlechat.go:147]:
```suggestion
```
nit
I think `prealloc` [is not enabled](https://github.com/grafana/grafana/blob/6c6a970f56845cccbcea9e8d581f0187553899f2/.golangci.toml#L29-L62) in our configuration
so this comment is not necessary

**Review by papagian — APPROVED** [review]

**Comment by papagian** [comment]
> @papagian who can help me with review?

approved! can you try to sync your branch with the latest grafana main (I cannot push in your branch) so that we make sure that all the CI checks succeed with the latest changes, thank you!

File: pkg/services/alerting/notifiers/googlechat.go (hunk 1 above)
Comment (peakle) [reply pkg/services/alerting/notifiers/googlechat.go:147]: yes, i want to enable it, so i split big pr https://github.com/grafana/grafana/pull/59118

**Pull request merged**